exporters/ostream/src/log_exporter.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/exporters/ostream/log_exporter.h"
+
+#include <iostream>
+
+namespace nostd   = opentelemetry::nostd;
+namespace sdklogs = opentelemetry::sdk::logs;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+OStreamLogExporter::OStreamLogExporter(std::ostream &sout) noexcept : sout_(sout) {}
+
+std::unique_ptr<sdklogs::Recordable> OStreamLogExporter::MakeRecordable() noexcept
+{
+  return std::unique_ptr<sdklogs::Recordable>(new sdklogs::LogRecord());
+}
+
+sdklogs::ExportResult OStreamLogExporter::Export(
+    const nostd::span<std::unique_ptr<sdklogs::Recordable>> &records) noexcept
+{
+  if (is_shutdown_)
+  {
+    return sdklogs::ExportResult::kFailure;
+  }
+
+  for (auto &record : records)
+  {
+    // Convert recordable to a LogRecord so that the getters of the LogRecord can be used
+    auto log_record =
+        std::unique_ptr<sdklogs::LogRecord>(static_cast<sdklogs::LogRecord *>(record.release()));
+
+    if (log_record == nullptr)
+    {
+      // Error "recordable data was lost"
+      continue;
+    }
+
+    // Convert trace, spanid, traceflags into exportable representation
+    constexpr int trace_id_len    = 32;
+    constexpr int span_id__len    = 16;
+    constexpr int trace_flags_len = 2;
+
+    char trace_id[trace_id_len]       = {0};
+    char span_id[span_id__len]        = {0};
+    char trace_flags[trace_flags_len] = {0};
+
+    log_record->GetTraceId().ToLowerBase16(trace_id);
+    log_record->GetSpanId().ToLowerBase16(span_id);
+    log_record->GetTraceFlags().ToLowerBase16(trace_flags);
+
+    // Print out each field of the log record, noting that severity is separated
+    // into severity_num and severity_text
+    sout_ << "{\n"
+          << "  timestamp     : " << log_record->GetTimestamp().time_since_epoch().count() << "\n"
+          << "  severity_num  : " << static_cast<int>(log_record->GetSeverity()) << "\n"
+          << "  severity_text : "
+          << opentelemetry::logs::SeverityNumToText[static_cast<int>(log_record->GetSeverity())]
+          << "\n"
+          << "  name          : " << log_record->GetName() << "\n"
```

Comment:
nit: `<< std::endl` instead of `<< "\n"`.

Reply:
Related- should traces' [span exporter](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/exporters/ostream/src/span_exporter.cc#L64) be updated as well? Or should we leave it here as `\n` to keept it consistent?

Reply:
Thanks for pointing it out, I think we could leave this to keep it consistent.